src/main/org/firebirdsql/management/FBStreamingBackupManager.java

```diff
@@ -0,0 +1,321 @@
+
+                for (int codePos = 0; codePos < buffer.length && buffer[codePos] != isc_info_end;) {
+                    switch (buffer[codePos]) {
+                    case isc_info_svc_stdin:
+                        int requestedBytes = Math.min(iscVaxInteger(buffer, ++codePos, 4), stdinBuffer.length);
+                        codePos += 4;
+                        if (requestedBytes > 0) {
+                            int actuallyReadBytes = restoreInputStream.read(stdinBuffer, 0, requestedBytes);
+                            if (actuallyReadBytes > 0) {
+                                infoSPB = service.createServiceParameterBuffer();
+                                if (stdinBuffer.length == actuallyReadBytes)
+                                    infoSPB.addArgument(isc_info_svc_line, stdinBuffer);
+                                else
+                                    infoSPB.addArgument(isc_info_svc_line,
+                                            Arrays.copyOfRange(stdinBuffer, 0, actuallyReadBytes));
+                            }
+
+                            restoreInputStream.mark(2);
+                            if (restoreInputStream.read() < 0)
+                                sending = false;
+                            else
+                                restoreInputStream.reset();
+                        }
+                        break;
+                    case isc_info_truncated:
+                        bufferSize *= 2;
+                        ++codePos;
+                        break;
+                    case isc_info_svc_line:
+                        int bytesToLog = readOutput(buffer, codePos, currentLogger);
+                        codePos += 3;
+                        switch (bytesToLog) {
+                        case -1:
+                            ++codePos;
+                            break;
+                        case 0:
+                            processing = false;
+                            break;
+                        default:
+                            codePos += bytesToLog;
+                            if (currentLogger != null)
+                                currentLogger.write(newLine);
+                        }
+                        break;
+                    case isc_info_end:
+                        break;
+                    default:
+                        throw new SQLException("Unexpected response from service. ");
+                    }
+                }
+            }
+        } catch (IOException ioe) {
+            throw new SQLException(ioe);
+        }
+    }
+
+    private int readOutput(byte[] buffer, int offset, OutputStream out) throws SQLException, IOException {
+        int dataLength = iscVaxInteger2(buffer, offset + 1);
+        if (dataLength == 0) {
+            switch (buffer[offset + 3]) {
+            case isc_info_data_not_ready:
+                return -1;
+            case isc_info_end:
+                return 0;
```

Comment:
Two things:

1. Create constants (eg `END_OF_FILE` and `DATA_NOT_READY`.
2. Could you switch the use of `0` and `-1`? Using `-1` for `isc_info_end` would match nicely with its meaning in streams.